helix-rest/src/main/java/org/apache/helix/rest/server/resources/zookeeper/ZooKeeperAccessor.java

```diff
@@ -196,6 +209,32 @@ private Response getStat(BaseDataAccessor<byte[]> zkBaseDataAccessor, String pat
     return JSONRepresentation(result);
   }
 
+  /**
+   * Delete the ZNode at the given path if exists.
+   * @param zkBaseDataAccessor
+   * @param path
+   * @return The delete result and the operated path.
+   */
+  private Response delete(BaseDataAccessor zkBaseDataAccessor, String path) {
```

Comment:
I'm not seeing how this is any different from using delete. This is no better than using `delete` for two different types of delete's - delete and deleteEphemeral.

Perhaps you could add a commandStr here to differentiate two different types of `delete`s, and when you want to add an endpoint for regular delete backed by ACL checks, then just implement that _if_ that becomes necessary?

My point was not about what kind of REST verb we should use - it's pretty clear we should use DELETE in this case. But it's more about following a good API design which, again, is something that is hard to misuse by not embedding hidden assumptions or TODOs that may cause a behavior change down the road.

Reply:
Discussed with Junkai in slack, his point is that we don't need the additional cmd layer for now.